src/dispatcher.cpp

```diff
@@ -84,8 +89,8 @@ void dispatcher::stop()
     {
         std::lock_guard< std::mutex > lock( _was_stopped_mutex );
         _was_stopped = true;
-        _was_stopped_cv.notify_all();
     }
+    _was_stopped_cv.notify_all();
```

Comment:
Do we still have the issue with the commented out code at the top of the function? Just curious...

Reply:
Yes, still causing issues.

I think it's happen because we return before we notify, and we got threads waiting on mutex and the dispatcher is destructed.
We may need to work on protecting all resources, with a sort of "invalidator" same as we did in the waiting_on